src/LibraryManager.Vsix/Commands/InstallLibraryCommand.cs

```diff
@@ -153,5 +158,34 @@ public static void Initialize(Package package, OleMenuCommandService commandServ
                 Telemetry.TrackUserTask("Open-InstallDialog");
             }
         }
+
+        private async Task<Manifest> GetManifestAsync(string configFilePath, IDependencies dependencies)
+        {
+            RunningDocumentTable rdt = new RunningDocumentTable(ServiceProvider.GlobalProvider);
+            IVsTextBuffer textBuffer = rdt.FindDocument(configFilePath) as IVsTextBuffer;
+
+            Manifest manifest = null;
+
+            // If the textBuffer is null, then libman.json is not open. In that case, we'll get the manifest from the the file.
+            // If textBuffer is not null, then libman.json file is open and could be dirty. So we'll get the contents for the manifest from the text buffer.
+            if (textBuffer == null)
+            {
+                manifest = await Manifest.FromFileAsync(configFilePath, dependencies, CancellationToken.None).ConfigureAwait(false);
+            }
+            else
+            {
+                IComponentModel componentModel = ServiceProvider.GlobalProvider.GetService(typeof(SComponentModel)) as IComponentModel;
+                IVsEditorAdaptersFactoryService editorAdapterService = componentModel.GetService<IVsEditorAdaptersFactoryService>();
+
+                ITextBuffer documentBuffer = editorAdapterService.GetDocumentBuffer(textBuffer);
+
+                if (documentBuffer != null)
```

Comment:
>if (documentBuffer != null) [](start = 16, length = 27)

Is this expected to be possible? If yes, then maybe we should use the Manifest.FromFileAsync code above

Reply:
I don't expect to hit this. I was only wondering if someone closed the document when operation was in progress.

I'll include the above code or refactor a bit

---
In reply to: [203158515](https://github.com/aspnet/LibraryManager/pull/323#discussion_r203158515) [](ancestors = 203158515)

Reply:
Change per what we did together.

---
In reply to: [203158905](https://github.com/aspnet/LibraryManager/pull/323#discussion_r203158905) [](ancestors = 203158905,203158515)

Reply:
@ToddGrun , I did not make the last change that we discussed, It would break this scenario:

1. libman.json file  is open and I remove a comma and cause syntax error
2. Manifest would be null in this case as can't parse correctly
3. The change that we made was to get the manifest using libman.json from disk which doesn't contain changes I made in first step. Manifest that I get using file from disk will parse file and show Add Client-Side libraries dialog instead of warning message box that something went wrong

I'll merge the changes as is and we can discuss tomorrow if you have any suggestion.